Revisions to GOVERNANCE.md to reflect the draft Tech Charter

bradcray opened the pull request:
This updates our GOVERNANCE.md file to reflect:
* the roles as defined by the technical charter that we've drafted with the Linux Foundation
* the Technical Steering Committee (TSC) that we've recently created
* and other recent changes in terms of personnel and roles
It was reviewed by the TSC between the meetings held on Sept 4 and Sept 18 and agreed upon at the Sept 18 meeting.  It is intended to be a living document that will continue to be updated and evolve as the project's governance continues to change and mature.

This version of the document refers to CHARTER.md, which is simply an incomplete placeholder at present.  My current understanding is that we'll receive a final copy of the technical charter for inclusion in this directory, likely in PDF form, in the coming days-to-weeks.

bradcray committed:
First draft revision to GOVERNANCE.md to reflect Tech Charter draft language

We recently received a proposed boilerplate Tech Charter for the Linux
Foundation which we imagine adopting in something close to its current
form.  This PR takes a pass over the GOVERNANCE.md document to try and
align it to that Tech Charter while also adding a file for the
technical charter to ultimately reside in.

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

brandon-neth commented on GOVERNANCE.md:
```diff
@@ -2,48 +2,87 @@ Governance of the Chapel Project
+implements its compiler, library modules, and runtime; and the
+project's documentation.  It was last updated in February 2025.  This
+file goes hand-in-hand with the project's technical charter,
+`CHARTER.md`, which defines the project's operations, IP polcy, and
```
polcy -> policy

brandon-neth commented on GOVERNANCE.md:
```diff
@@ -2,48 +2,87 @@ Governance of the Chapel Project
+If you have specific requests with respect to how the project is
+governed, please feel encouraged to capture those in an issue on this
+repository, or to bring it up [in our community
+fprums](https://chapel-lang.org/community/).
```
fprums -> forums

brandon-neth commented on GOVERNANCE.md:
```diff
@@ -2,48 +2,87 @@ Governance of the Chapel Project
+* Contributors, who have contributed technical artifacts into the
+  Chapel release
+* Community Collaborators (those who contribute in other ways, notably
+  through feature requests, bug reports, and feedback.
```
missing end paren

brandon-neth commented:
Other than the typos above, I think this looks incredible.

bradcray committed:
Fix typos pointed out by Brandon (and ran a spellcheck to try to find others)

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

bradcray committed:
Clarify that contributors also include those who triage and develop release artifacts

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

bradcray committed:
Additional edits to GOVERNANCE document based on last few months

These reflect changes to the composition of the subteam leads and
committers, as well as where we stand w.r.t. the TSC at present.

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

bradcray committed:
Update Anna's name to reflect the fact that I'd already updated the URL to point to her

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

lydia-duncan commented on GOVERNANCE.md:
```diff
@@ -2,57 +2,127 @@ Governance of the Chapel Project
-**Subteam Leads**
+project, covering the language's definition; the code base that
+implements its compiler, library modules, runtime, and tools; and the
+project's documentation.  It was last updated in February 2025.  This
```
You'll want to update this date again :)

bradcray replied:
Good catch, thanks!

lydia-duncan commented on GOVERNANCE.md:
```diff
@@ -2,57 +2,127 @@ Governance of the Chapel Project
+community:
+
+* the Technical Steering Committee (TSC) and its Chair
+* TSC Chair
```
I think this is redundant with the previous bullet

lydia-duncan reviewed (approved):
Just a couple of minor nits

bradcray committed:
Fix a couple of issues that Lydia caught

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

bradcray committed:
Merge branch 'main' of https://github.com/bradcray/chapel-governance into tech-charter

agirish reviewed (approved):
Looks good to me!

stonea commented on GOVERNANCE.md:
```diff
@@ -157,15 +182,20 @@ artifacts.  Such contributions are typically made by directly opening
+[Developer Certificate of Origin
 (DCO)](https://github.com/chapel-lang/chapel/blob/main/.github/CONTRIBUTING.md).
 
 As noted above, such PRs must be reviewed and merged by a core
```
You don't define what is meant by "core contributor".  I'm assuming it's synonymous with "committer" as defined above.

bradcray replied:
Good catch, that's old terminology that I've become blind to.  I fixed one other related mention of "core" as well in the latest commits (the second of which was a spelling/formatting fix).

bradcray committed:
Update a few reference to "core contributors" that should be "Committers"

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

bradcray committed:
Spellcheck

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

e-kayrakli commented on GOVERNANCE.md:
```diff
@@ -2,57 +2,126 @@ Governance of the Chapel Project
-particularly for those that are contentious, represent a significant
-change in direction, or have the possibility of impacting other
-subteams.
+but are expected to keep the TSC, through its chair, aware of major
```
This assumes a chair. maybe "through the Tech Lead"? Even if we decide to have a TSC Chair role, Tech Lead seems to be the better role for this type of liaison to me.

e-kayrakli reviewed (approved):
I just picked on a semi-technicality. Other than that, looks good.

bradcray committed:
s/chair/tech lead for subteam leads informing the TSC of big things

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

ShreyasKhandekar commented on GOVERNANCE.md:
```diff
@@ -2,57 +2,126 @@ Governance of the Chapel Project
+Generally speaking, all committers are contributors, and any
+contributor or community collaborator can request to join the TSC.  At
+present, we are lacking a means for a community members to be a
+contributor, but one of the roles of the TSC will be to define this
```
Did you mean committer here?
```suggestion
committer, but one of the roles of the TSC will be to define this
```

bradcray committed:
Update GOVERNANCE.md

Fix committer/contributor mix-up that Shreyas caught.  Those words are too similar for my brain, apparently.

Co-authored-by: Shreyas Khandekar <60454060+ShreyasKhandekar@users.noreply.github.com>

arifthpe commented on GOVERNANCE.md:
```diff
@@ -65,87 +134,43 @@ These team members are expected to participate in reviewing PRs for
+help out.
 
 At present, the list of committers includes all of the
 technical leaders listed previously, as well as:
```
I think it would be better to list _all_ the committers here even if some are also the tech lead or subteam leads. Otherwise if leads change, this list is a second location needing update and could get missed.

bradcray replied:
I think that's a good thought.  I was being needlessly lazy.

agirish replied:
I had same thoughts as @arifthpe  - as the subset of names had me confused for a second. I figured out what Brad was trying to do, so didn't bother commenting on it. But agree that listing all committers here will make it more clear. But even better might be if don't list anyone in there and move list of ALL committers to a separate COMMITTERS.md file. As that list should ideally grow over time and shouldn't need updating the main governance doc everytime.

bradcray replied:
I'm not necessarily opposed to this idea, but am going to proceed with what we have for now since it's what the TSC reviewed and voted on.  I'd also be interested in surveying more LF/HPSF projects to see how they approach this, to try and keep a consistent organization.  I'll note that one advantage to a separate file is that if we were to auto-tag issues as being from "the community" rather than a committer (similar to our "external issue" / "user issue" tags today), this file would be a natural way to distinguish those categories.

arifthpe commented on GOVERNANCE.md:
```diff
@@ -65,87 +134,43 @@ These team members are expected to participate in reviewing PRs for
+help out.
 
 At present, the list of committers includes all of the
 technical leaders listed previously, as well as:
```
If we leave it just listing those who aren't leads, "technical leaders" should probably be changed to "Tech Lead and Subteam Leads" to be unambiguous

bradcray committed:
Avoid relying on inheritance when specifying committers

---
Signed-off-by: Brad Chamberlain <bradcray@users.noreply.github.com>

bradcray committed:
Merge branch 'tech-charter' of https://github.com/bradcray/chapel-governance into tech-charter
